testonly/setup/trees.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/crypto/keyspb"
+	"github.com/google/trillian/crypto/sigpb"
+	"github.com/google/trillian/testonly"
+
+	ktestonly "github.com/google/trillian/crypto/keys/testonly"
+)
+
+// CreateLog creates a Trillian log with for testing purposes
+//
+// It requires the caller to specify a maxRootDuration for the Trillian tree,
+// which will be used as the interval after which a new signed root is produced
+// despite no submissions.
+func CreateLog(ctx context.Context, t *testing.T, logClient trillian.TrillianLogClient, adminClient trillian.TrillianAdminClient, maxRootDuration time.Duration) *trillian.Tree {
```

Comment:
It would be nice to avoid an explosion of test helper functions, to set all the various fields of `trillian.Tree`. 

Would it be useful to create a `TreeOption` construct like GRPC's [`With*`](https://github.com/grpc/grpc-go/blob/master/dialoptions.go#L273) methods?